### ISSUE TITLE
Question about GRU module

### ISSUE THREAD

**Issue by Hannibal046** [shenwzh3/DAG-ERC#3]
Hello,
Thanks for your work on ERC. After reading the paper, I got a question about the GRU module, since all inputs to GRU is of size `[batch_size,1,hid_dim]`, which means it did not take sequential information into the model  due to the `seq_len = 1`, am I right? Did you try different architectures?
Thanks for your help !

**Comment by shenwzh3** [comment]
Hi, you can check line 112-119 in model.py. We use a for loop to process each utterance iteratively, in this architecture each utterance recieves the hidden states of all its predecessors, so the sequential information is considered.

**Comment by Hannibal046** [comment]
Hi, thanks for response.
I notice that in your paper, the input to GRU is the aggregated message from previous utterances, so do you mean this is your consideration of sequential information, even considering input to GRU of length 1?

And how to update the first utterance's repersentation since it does not have any predecessors ?

**Comment by shenwzh3** [comment]
> I notice that in your paper, the input to GRU is the aggregated message from previous utterances, so do you mean this is your consideration of sequential information, even considering input to GRU of length 1?

Yes , for each layer, each utterance is updated iteratively in the same GRU unit, that works similarly to the traditional GRU recurrent networks. Except for the only difference that in our network at each time step the utterance reveive all the predecessors' hidden state, while in the traditional GPU recurrent networks only the t-1's hidden state is received.


> And how to update the first utterance's repersentation since it does not have any predecessors ?

In our implementation the first utterance's predecessor is set to 0, it works just like that in RNN, the initial state h_0 is 0.
